core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -614,6 +623,33 @@ public void handleDeclarativeServices(AiServicesRecorder recorder,
         }
     }
 
+    @BuildStep
+    public void markUsedOutputGuardRailsUnremovable(List<AiServicesMethodBuildItem> methods,
+            BuildProducer<UnremovableBeanBuildItem> unremovableProducer) {
+        for (AiServicesMethodBuildItem method : methods) {
+            List<String> list = method.getGuardrails();
+            for (String cn : list) {
+                unremovableProducer.produce(UnremovableBeanBuildItem.beanTypes(DotName.createSimple(cn)));
+            }
+        }
+    }
+
+    @BuildStep
+    public void detectMissingGuardRails(SynthesisFinishedBuildItem synthesisFinished,
+            List<AiServicesMethodBuildItem> methods,
+            BuildProducer<ValidationPhaseBuildItem.ValidationErrorBuildItem> errors) {
+
+        for (AiServicesMethodBuildItem method : methods) {
+            List<String> list = method.getGuardrails();
+            for (String cn : list) {
+                if (synthesisFinished.beanStream().withBeanType(DotName.createSimple(cn)).isEmpty()) {
+                    errors.produce(new ValidationPhaseBuildItem.ValidationErrorBuildItem(
+                            new DeploymentException("Missing guardrail bean: " + cn)));
+                }
+            }
+        }
+    }
```

Comment:
Doesn't Arc automatically do this?

Reply:
You get a runtime error when you try to access it (we resolve it using a programmatic lookup). Thus, it does not give you the error at build time.

Reply:
Interesting

Reply:
(remove that code, tests will fail :-))